libs/execution/src/lib/types/valuetypes/standard-representation.ts

```diff
@@ -18,21 +18,22 @@ export class StandardRepresentationResolver {
   constructor(private value: unknown) {}
 
   fromValuetype(valuetype: Valuetype): PrimitiveType {
-    if (valuetype instanceof BooleanValuetype) {
+    if (valuetype === PrimitiveValuetypes.Boolean) {
       return this.fromBooleanValuetype(valuetype);
-    } else if (valuetype instanceof DecimalValuetype) {
+    } else if (valuetype === PrimitiveValuetypes.Decimal) {
       return this.fromDecimalValuetype(valuetype);
-    } else if (valuetype instanceof IntegerValuetype) {
+    } else if (valuetype === PrimitiveValuetypes.Integer) {
       return this.fromIntegerValuetype(valuetype);
-    } else if (valuetype instanceof TextValuetype) {
+    } else if (valuetype === PrimitiveValuetypes.Text) {
       return this.fromTextValuetype(valuetype);
     } else if (valuetype instanceof AtomicValuetype) {
       return this.fromAtomicValuetype(valuetype);
     }
     throw Error('Parsing from unsupported value type');
   }
 
-  fromBooleanValuetype(valuetype: BooleanValuetype): boolean {
+  fromBooleanValuetype(valuetype: PrimitiveValuetype): boolean {
+    assert(valuetype === PrimitiveValuetypes.Boolean);
```

Comment:
In general, I'd prefer static checks by the compiler over runtime checks via `assert`. Is there a problem with the `BooleanValuetype` as parameter?

Reply:
The class `BooleanValuetype` is currently not exported (to enforce Singleton). Thus, `PrimitiveValuetype` is used here and in visitors. I can try to export the interface/type of the class and introduce type-guards to fix this while leaving the implementation class private to the file.

Reply:
Looks good 👍 in case you have trouble with that, your solution with `assert` is also fine. Or alternatively remove the parameter entirely and access the valuetype instance directly via `PrimitiveValuetypes.Boolean`.

Reply:
Addressed in [d5d2c3b](https://github.com/jvalue/jayvee/pull/317/commits/d5d2c3b6773e892a9b0b9abd61f2a9e94f82534c)